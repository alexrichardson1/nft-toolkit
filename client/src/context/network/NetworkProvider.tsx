import { useState } from "react";
import NetworkContext from "./NetworkContext";

interface PropsT {
  children: React.ReactNode;
}

const NetworkProvider = (props: PropsT): JSX.Element => {
  const [selectedNet, setSelectedNet] = useState("Ethereum");
  return (
    <NetworkContext.Provider value={{ selectedNet, setSelectedNet }}>
      {props.children}
    </NetworkContext.Provider>
  );
};

export default NetworkProvider;
