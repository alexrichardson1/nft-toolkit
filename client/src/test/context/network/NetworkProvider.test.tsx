import NetworkProvider from "context/network/NetworkProvider";
import { mount } from "enzyme";

describe("NetworkProvider unit tests", () => {
  afterAll(() => localStorage.clear());

  test("preferred network set in local storage", () => {
    localStorage.setItem("preferred-network", "ethereum");
    expect(mount(<NetworkProvider>Test</NetworkProvider>)).toMatchSnapshot();
  });
});
