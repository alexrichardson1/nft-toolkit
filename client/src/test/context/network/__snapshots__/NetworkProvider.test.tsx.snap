// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkProvider unit tests preferred network set in local storage 1`] = `
<NetworkProvider>
  Test
</NetworkProvider>
`;
